powerauth-push-server/src/main/java/io/getlime/push/errorhandling/DefaultExceptionHandler.java

```diff
@@ -103,4 +104,19 @@ public ErrorResponse handlePowerAuthClientException(Exception e) {
         logger.error(e.getMessage(), e);
         return new ErrorResponse(WebServiceError.Code.ERROR_PA_SERVER_COMM, e);
     }
+
+    /**
+     * Handle exceptions related to inbox.
+     * @param e {@link InboxMessageNotFoundException}
+     * @return Error response.
+     */
+    @ResponseStatus(HttpStatus.NOT_FOUND)  // 404
```

Comment:
I am not sure about the `404`, it will be also returned when the application is not deployed. We can keep it, let's just be aware of this.

Reply:
I will change this to 400...